gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java

```diff
@@ -280,17 +227,41 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
         return (GQuery) o;
       }
       if (o instanceof Function) {
-        return $((Function) o);
-      }
-      if (JsUtils.isElement(o)) {
-        return $(JsUtils.<Element> cast(o));
-      }
-      if (o instanceof JavaScriptObject) {
-        return $((JavaScriptObject) o);
+        return $(((Function) o).getElement());
```

Comment:
Will take a look to all possibilities. The problem is that this seems not fix the issue of the user, I cannot reproduce his problem though. Let see if I reach a solution to have the issue fixed. Then we can focus on reviewing this code.